### ISSUE TITLE
function hdb_catalog.notify_hasura_comments_UPDATE() does not exist

### ISSUE THREAD

**Issue by loremru** [hasura/graphql-engine#9270]
### VERSIONS:

postgres:
    image: postgres:12
graphql-engine:
    image: hasura/graphql-engine:v2.10.1

### ERROR:
WARN Metadata is inconsistent
WARN Use 'hasura metadata ic list' command to list inconsistent objects 
ERRO skipping applying migrations on database gunza, encountered:
{
  "code": "postgres-error",
  "error": "query execution failed",
  "internal": {
    "arguments": [],
    "error": {
      "description": null,
      "exec_status": "FatalError",
      "hint": null,
      "message": "function hdb_catalog.notify_hasura_comments_UPDATE() does not exist",
      "status_code": "42883"
    },
    "prepared": false,
    
   Error occures when i trying to deploy my metadata/migrations to new clear hasura with same version. On the current hasura migrations are set properly

![image](https://user-images.githubusercontent.com/66596500/205891889-04882f19-4f65-4757-81c8-1505a67d70da.png)

**Comment by BenoitRanque** [comment]
Have you tried reloading metadata? It causes triggers associated with event triggers to be recreated and can help with these kinds of problems

**Comment by loremru** [comment]
@BenoitRanque
> Have you tried reloading metadata? It causes triggers associated with event triggers to be recreated and can help with these kinds of problems

Yes, I did. I tried to recreate metadata/migrations, upload them one more time, but nothing helps. Same error.

**Comment by thecoolestcoderrushil** [comment]
deploying metadata and migrations to a fresh hasura that fails with function hdb_catalog.notify_hasura_comments_update does not exist is a painful blocker, since the catalog trigger is missing on a clean database. rehearsing the migration on a production-shaped twin first would surface the catalog problem before deploy day. antifailure.dev (https://antifailure.dev) builds disposable database twins for migration rehearsal and is open source and free to try, with demos available for teams that want one.
